### PR TITLE
Added `finetune_cnn_jax.ipynb` for `finetune_cnn_torch.ipynb`

### PR DESCRIPTION
**Pull request by susnato**
## Description

Converted `finetune_cnn_torch.ipynb` to  `finetune_cnn_jax.ipynb` .

### Colab link
https://colab.research.google.com/drive/1c0yus2G9AIHXjstUDGT9u7cXgAkJ4tCF?usp=sharing

### Issue 

https://github.com/probml/pyprobml/issues/686

## Checklist:

- [✔] Performed a self-review of the code
- [✔] Tested on Google Colab.

### CODE REVIEW

**Comment by susnato** [comment]
Can you please review my code? @murphyk @mjsML @gerdm

**Comment by susnato** [comment]
Currently, the notebook is being run for some small changes so the viewers might not be able to see the outputs using the Colab link.

File: notebooks-d2l/finetune_cnn_jax.ipynb [hunk 1]
```diff
@@ -0,0 +1,816 @@
+{
```
Comment (murphyk) [inline notebooks-d2l/finetune_cnn_jax.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #10.        x = nn.BatchNorm(use_running_average=True)(x)
</pre><p>Should there be a train/test flag to decide if <code>use_running_average</code> is True or False? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/probml-notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='16'/>

Comment (murphyk) [inline notebooks-d2l/finetune_cnn_jax.ipynb:1]: <p>Training is slow - is there a way to speed it up? (Larger batch size?)</p><p>Is there a way to silence this annoying warning?</p><pre class="ql-syntax" spellcheck="false">Using cache found in /root/.cache/torch/hub/pytorch_vision_v0.10.0
<code></code>`
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/probml/probml-notebooks/pull/68/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

**Comment by murphyk** [comment]
Data loading / augmentation should not take 15 minutes!  Please investigate.

**Comment by susnato** [comment]
> Data loading / augmentation should not take 15 minutes! Please investigate.

I loaded the full data into memory that's why the data loading is taking 15 minutes, I will use TF data loader and will get back to you as soon as possible.

File: notebooks-d2l/finetune_cnn_jax.ipynb (hunk 1 above)
Comment (susnato) [reply notebooks-d2l/finetune_cnn_jax.ipynb:1]: If I increase the batch size (upto 32) then after some iterations the notebook will go out of GPU memory and give out error, that's why I preferred to keep it low, but I will surely investigate on this.

This annoying warning is totally related to the external repository I used, I will change the model and see if I am still getting this warning or not.

**Comment by susnato** [comment]
Hi, @murphyk I made some changes,

1. I added a data loader, now there is no need to load the data beforehand (which used to take 15 minutes!).
2. I removed the annoying warning `Using cache found in /root/.cache/torch/hub/pytorch_vision_v0.10.0` (it will only show once or twice at the start of the training ).
3. I made the training faster. (training for 1 epoch and validation is now taking 3-4 minutes ) .
4. Now we can take up to 64 images per batch whereas previously it was only 16 (btw use `batch_size=32` in the data loader because it will take 1 pair of hotdog and non-hotdog per example so for 32 it will take 64 images).

Please let me know if any more changes are needed or not.
